src/commands/cmd_dispatcher.c

```diff
@@ -228,7 +228,8 @@ int CommandDispatch(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
 	ExecutorThread exec_thread = (flags & (REDISMODULE_CTX_FLAGS_MULTI         |
 										   REDISMODULE_CTX_FLAGS_LUA           |
 										   REDISMODULE_CTX_FLAGS_DENY_BLOCKING |
-										   REDISMODULE_CTX_FLAGS_LOADING)) ?
+										   REDISMODULE_CTX_FLAGS_LOADING)) || 
+										   is_replicated ?
 								 EXEC_THREAD_MAIN : EXEC_THREAD_READER;
```

Comment:
I think `is_replicated` might be more likely, therefor let's switch the conditions:
```c

bool main_thread = (is_replicated || (flags & (REDISMODULE_CTX_FLAGS_MULTI         |
 										   REDISMODULE_CTX_FLAGS_LUA           |
 										   REDISMODULE_CTX_FLAGS_DENY_BLOCKING |
 										   REDISMODULE_CTX_FLAGS_LOADING));
ExecutorThread exec_thread = (main_thread) ? EXEC_THREAD_MAIN : EXEC_THREAD_READER
```